modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java

```diff
@@ -743,6 +743,25 @@ public void run() {
         StatusThreadPoolExecutor.getInstance().execute(r);
     }
 
+    @ReactMethod
+    public void signTypedData(final String data, final String password, final Callback callback) {
+        Log.d(TAG, "signTypedData");
+        if (!checkAvailability()) {
+            callback.invoke(false);
+            return;
+        }
+
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                String res = Statusgo.signTypedData(data, password);
+                callback.invoke(res);
+            }
+        };
+
+        StatusThreadPoolExecutor.getInstance().execute(r);
```

Comment:
Could be simplified by using an [inline lambda](https://www.oracle.com/webfolder/technetwork/tutorials/obe/java/Lambda-QuickStart/index.html)

Reply:
@jeluard I just used code from other methods we have, do you think we need to change how it works in other methods? maybe we could do refactoring in a separate issue?

Reply:
Right, separate issue :D